### PR TITLE
feat: Modal support hooks

### LINKED ISSUE #20177: 4.0 Modal.xxx use hooks
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### What problem does this feature solve?
use hooks to support configProvider.

### What does the proposed API look like?
```tsx
import { Modal } from 'antd';

function Demo() {
  const modal = useModal();
  
  React.useEffect(() => {
    modal.info({
      // ...
    });
  }, []);
}
```

hooks:
```tsx
function useModal() {
  const context = React.useContext(ConfigProviderContext);
  // ...
}
```

### PR DESCRIPTION
**Pull request by zombieJ**
### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

close #20177

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Modal add `useModal` hook to support context access.      |
| 🇨🇳 Chinese |      Modal 添加 `useModal` hook 以支持 context 访问。     |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed
- [ ] Test case

-----
[View rendered components/modal/demo/hooks.md](https://github.com/ant-design/ant-design/blob/modal-hooks/components/modal/demo/hooks.md)

### CODE REVIEW

**Comment by zombieJ** [comment]
Please help to review.

**Comment by zombieJ** [comment]
ping

**Review by afc163 — APPROVED** [review]

File: components/modal/demo/hooks.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
+---
+order: 12
+title:
+  zh-CN: 使用 hooks
```
Comment (afc163) [inline components/modal/demo/hooks.md]: 这里最好写成解决 context 问题的标题。

Comment (afc163) [reply components/modal/demo/hooks.md]: 使用 hooks 不是关键词。

**Pull request merged**